### DISCUSSION TITLE
ESP01S: Not enough space

### DISCUSSION THREAD

**Discussion by Andre0be** [arendst/Tasmota#18566] [Q&A]
Hello,

I have an ESP01S (flash 1MB) with Tasmota 12.5.0.1 and the file system hat not enough space.

Upload Failed  Not enough space - is the message.

I build the image with IPv6 Support, use script, use SML_M, use rules. Not more.

The image is packed 516kb, unpacked 732kb. Thats to much for ESP01s. Example the webinterface is not displayed.

What function can be disable, to build a smaler image?

Image with version 12.4.0.5 works fine. With 12.5 seems to have become a lot bigger.

**Comment by Jason2866** [comment]
There is NO file system with the 1M build. You need a module with 4 MB flash and have to use `tasmota-4M`

**Reply by sfromis** [reply]
Your size of 732KB is not "too much" for ESP-01, but for upgrading you should always use the compressed `.bin.gz` image, and when not doing it from a web server, you should manually do an extra step flashing `tasmota-minimal.bin.gz` first. If still having size trouble, use wired flashing instead of via the web UI.

**Comment by Andre0be** [comment]
Flash Size: 1024 KB
Programm Flash Size : 1024 KB
Programm Size: 718 KB

this is what I get displayed.

**Comment by Jason2866** [comment]
Yes, a big firmware build. Remove unneeded driver and sensors and it will shrink.
For what is IPv6 used?
Btw you can't build a firmware using Rules and Script. There will be just one of them in the firmware.

**Comment by Andre0be** [comment]
I have a dual stack lite connection. I need IPv6.

Can you tell me, which drivers and sensors i can disable? and how? Thx

**Comment by Jason2866** [comment]
How are the ISP connection IP protocols related to the internal LAN IP protocol?
Easiest to safe flash space is to disable unneeded IPv6 protocol. 

https://tasmota.github.io/docs/Compile-your-build/
